numerics/polynomial_in_monomial_basis.hpp

```diff
@@ -145,248 +151,199 @@ class PolynomialInMonomialBasis : public Polynomial<Value_, Argument_> {
       not_null<serialization::Polynomial*> message) const override;
   static PolynomialInMonomialBasis ReadFromMessage(
       serialization::Polynomial const& message);
+  template<template<typename, typename, int> typename Evaluator_>
+  static PolynomialInMonomialBasis ReadFromMessage(
+      serialization::Polynomial const& message,
+      with_evaluator_t<Evaluator_>);
 
  private:
+  static constexpr Evaluator<Value_, Difference<Argument_>, degree_> const*
+  DefaultEvaluator();
+
   Coefficients coefficients_;
   Argument origin_;
-
-  template<typename V, typename A, int r,
-           template<typename, typename, int> typename E>
-  constexpr PolynomialInMonomialBasis<V, A, r, E>
-  friend operator-(PolynomialInMonomialBasis<V, A, r, E> const& right);
-  template<typename V, typename A, int l, int r,
-           template<typename, typename, int> typename E>
-  constexpr PolynomialInMonomialBasis<V, A, PRINCIPIA_MAX(l, r), E>
-  friend operator+(PolynomialInMonomialBasis<V, A, l, E> const& left,
-                   PolynomialInMonomialBasis<V, A, r, E> const& right);
-  template<typename V, typename A, int l, int r,
-           template<typename, typename, int> typename E>
-  constexpr PolynomialInMonomialBasis<V, A, PRINCIPIA_MAX(l, r), E>
-  friend operator-(PolynomialInMonomialBasis<V, A, l, E> const& left,
-                   PolynomialInMonomialBasis<V, A, r, E> const& right);
+  Evaluator<Value_, Difference<Argument_>, degree_> const* evaluator_;
```

Comment:
Add a setter for `evaluator_`; we may want to set it on the result of an operator `+`, or after applying a mutating operation such as `+=`.